db/boltdb.go

```diff
@@ -50,11 +50,13 @@ func NewBoltDB(dbPathname string) (*BoltDB, error) {
 	return &BoltDB{db: db}, err
 }
 
-func (d *BoltDB) Load(ident string, data interface{}) error {
-	return d.db.View(func(tx *bolt.Tx) error {
+func (d *BoltDB) Load(ident string, data interface{}) (bool, error) {
+	found := true
+	return found, d.db.View(func(tx *bolt.Tx) error {
 		b := tx.Bucket(topBucket)
 		v := b.Get([]byte(ident))
 		if v == nil {
+			found = false
```

Comment:
you are making a rather big assumption about evaluation here, namely that the `found` in `return found, ...` is evaluated *after* the `View` function has executed. The go spec may well guarantee that, but I'd rather avoid such subtlety.